doc/memos/rdm-draft-petry-riot-design-goals.md

```diff
@@ -0,0 +1,330 @@
+
+ - rdm-draft-petry-riot-design-goals.md
+ - Title: RIOT design goals
+ - Author: Daniel Petry
+ - Date: October 2018
+
+## Abstract
+
+This document presents key design aspects of the RIOT operating system: its
+application contexts and design rationales.  It aims to reflect a consensus on
+the reasoning behind RIOT design, in order to help new contributors get up to
+speed more quickly and provide common ground for technical debates.
+
+## Status
+
+The content of this document is licensed with a Creative Commons CC-BY-SA
+license.
+
+## Terminology
+
+Throughout this document, the term "users" refers to developers who are using
+RIOT as a basis to implement application software, without editing RIOT's
+source code. The term "developers" refers to contributors to the source code of
+RIOT.
+
+# 1. Introduction
+
+This document lists a set of generalised, loose requirements for RIOT (in a
+voluntary open source context, gathering strict requirements is neither
+possible nor welcome). These "design goals", as they may be called, are given
+as a set of use cases followed by a set of design philosophies. These
+correspond loosely to user requirements and derived requirements respectively.
+
+# 2. Use cases
+
+RIOT is a general purpose IoT operating system for low-end devices. As a
+result, its use cases are varied: below is a comprehensive, but non-exclusive
+list of them, including the general requirements placed on the devices for each
+use case.
+
+## 2.1. Environmental sensing
+
+Networks of remote sensor nodes can be deployed to monitor the environment or
+assist with urban planning. These devices can be distributed on their own or
+fit to infrastructure. The devices need to:
+
+  - Sense air composition, temperature, light quality, water quality, etc
+  - Support low-power wireless networking protocols
+  - Interoperate with non-RIOT nodes or gateways
+  - Collect data on timescales of the order of hours or longer
+  - Potentially be able to send data over long ranges
+  - Potentially communicate with a smartphone
+  - Potentially operate for years without power infrastructure or maintenance
+
+## 2.2. Rapid prototyping, research, and experimentation
+
+In experimentation and hacking situations, development needs to be easily
+accessible, and allow a short development time and quick results. This means
+that the software and hardware should:
+
+  - Let users prototype easily
+  - Let users easily run and load simple applications
```

Comment:
what does "run and load" mean? There's no runtime loading of applications (unless lua or jerryscript is used).

Reply:
Maybe this should be the other way around? It wasn't meant to imply runtime loading so this change might clarify